### ISSUE TITLE
cmake fails when configured with -pie

### ISSUE THREAD

**Issue by stanfordcox** [dyninst/dyninst#812]
**Intention**
Red Hat and Fedora builds are done with -pie.  The cmake machinery fails with -pie due to a compile/link incompatibility in dyninstAPI_RT/CMakeLists.txt

**Describe the bug**
This change to dyninstAPI_RT/CMakeLists.txt fixes it

```
@@ -5,4 +5,4 @@
 set (DYNINST_ROOT ${PROJECT_SOURCE_DIR}/..)
-set(CMAKE_C_FLAGS "${DYNINST_RT_CMAKE_C_FLAGS}")
-set(CMAKE_CXX_FLAGS "${DYNINST_RT_CMAKE_CXX_FLAGS}")
+# set(CMAKE_C_FLAGS "${DYNINST_RT_CMAKE_C_FLAGS}")
+# set(CMAKE_CXX_FLAGS "${DYNINST_RT_CMAKE_CXX_FLAGS}")

```
<root-dir>/CMakeLists.txt does
```
set(DYNINST_RT_CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
set(DYNINST_RT_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

but near as I can tell that never gets pushed out anywhere so later at:
```
if(BUILD_RTLIB)
  # Build the RT library as a separate project so we can change compilers
```

So we end up with an incompatibility between the C_FLAGS and LD_FLAGS so the test compile is done without -pie and the link is attempted with -pie and it fails.

Is DYNINST_RT_CMAKE_C_FLAGS needed?

**To Reproduce**
Fedora has rpm machinery that pulls in standard build flags.  So it is essentially building with:
O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection 

where redhat-hardened-cc1 is
*cc1_options:
+ %{!r:%{!fpie:%{!fPIE:%{!fpic:%{!fPIC:%{!fno-pic:-fPIE}}}}}}

**Comment by hainest** [comment]
>Is DYNINST_RT_CMAKE_C_FLAGS needed?

I'm not sure. I've always been suspicious of why the runtime would need to be built with a different compiler. I think now is as good a time as any to dig into that. Thanks for the report! Are you able to get things working on your end for now?

**Comment by stanfordcox** [comment]
On 8/25/20 17:39, Tim Haines wrote:

> Are you able to get things 
> working on your end for now?

The build works fine now.  Just working through fedora changes to the 
cmake macros then I will have the Fedora build done.  Then I'll do a 
pull request for my cmake changes.  It seems like all the machinery to 
cmake and build dyninstAPI_RT could go away but I took a way simpler 
approach.

**Comment by stanfordcox** [comment]
https://github.com/dyninst/dyninst/pull/817

**Comment by hainest** [comment]
Closed by #817.
